todo/android/todo/src/main/java/com/mongodb/stitch/sdk/examples/todo/MainActivity.java

```diff
@@ -323,170 +303,131 @@ public void onComplete(@NonNull final Task<Document> task) {
-                                            initTodoView();
+                                        public void onComplete(@NonNull final Task<StitchUser> task) {
+                                            if (task.isSuccessful()) {
+                                                initTodoView();
+                                            } else {
+                                                Log.e(TAG, "Error logging in with Facebook", task.getException());
+                                            }
                                         }
                                     });
-                            LoginManager.getInstance().logInWithReadPermissions(
-                                    MainActivity.this,
-                                    Arrays.asList("public_profile"));
-                        }
-                    });
-                    findViewById(R.id.fb_login_button_frame).setVisibility(View.VISIBLE);
-                }
-
-                if (info.hasGoogle()) {
-                    final GoogleSignInOptions.Builder gsoBuilder = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
-                            .requestServerAuthCode(info.getGoogle().getConfig().getClientId(), false);
-                    final GoogleSignInOptions gso = gsoBuilder.build();
+                                }
 
-                    if (_googleApiClient != null) {
-                        _googleApiClient.stopAutoManage(MainActivity.this);
-                        _googleApiClient.disconnect();
-                    }
+                                @Override
+                                public void onCancel() {}
 
-                    _googleApiClient = new GoogleApiClient.Builder(MainActivity.this)
-                            .enableAutoManage(MainActivity.this, new GoogleApiClient.OnConnectionFailedListener() {
                                 @Override
-                                public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
-                                    Log.e(TAG, "Error connecting to google: " + connectionResult.getErrorMessage());
+                                public void onError(final FacebookException exception) {
+                                    initTodoView();
                                 }
-                            })
-                            .addApi(GOOGLE_SIGN_IN_API, gso)
-                            .build();
+                            });
+                    LoginManager.getInstance().logInWithReadPermissions(
+                            MainActivity.this,
+                            Arrays.asList("public_profile"));
+                }
+            });
+            findViewById(R.id.fb_login_button_frame).setVisibility(View.VISIBLE);
+        }
 
-                    findViewById(R.id.google_login_button).setOnClickListener(new View.OnClickListener() {
+        if (!googleClientId.equals("TBD")) {
```

Comment:
Maybe this can throw instead?

Reply:
I'd rather make specific authentication methods optional for people to implement, than require them to go through the entire setup rigamarole.

That said, I'm flexible here

Reply:
That's fair. Carry on